pytest-backend/exasol/pytest_backend/itde_config.py

```diff
@@ -0,0 +1,236 @@
+from collections import ChainMap
+from dataclasses import dataclass
+from typing import (
+    Generic,
+    Optional,
+    TypeVar,
+)
+
+from exasol_integration_test_docker_environment.lib.test_environment.ports import Ports
+
+T = TypeVar("T")
+
+
+@dataclass(frozen=True)
+class ItdeOption(Generic[T]):
+    name: str
+    prefix: str
+    type: T
+    default: Optional[T] = None
+    help_text: str = ""
+
+    @property
+    def env(self):
+        """Environment variable name"""
+
+        def normalize(name):
+            name = name.replace("-", "_")
+            name = name.upper()
+            return name
+
+        return f"{normalize(self.prefix)}_{normalize(self.name)}"
+
+    @property
+    def cli(self):
+        """Cli argument name"""
+
+        def normalize(name):
+            name = name.replace("_", "-")
+            name = name.lower()
+            return name
+
+        return f"--{normalize(self.prefix)}-{normalize(self.name)}"
+
+    @property
+    def pytest(self):
+        """Pytest option name"""
+
+        def normalize(name):
+            name = name.replace("-", "_")
+            name = name.lower()
+            return name
+
+        return f"{normalize(self.prefix)}_{normalize(self.name)}"
+
+    @property
+    def help(self):
+        """Help text including information about default value."""
+        if not self.default:
+            return f"{self.help_text}."
+        return f"{self.help_text} (default: {self.default})."
+
+
+class ItdeOptionGroup:
```

Comment:
Don't see why to make `OptionGroup` Itde dependent.
One could simply preserve the generic functionality while adding the Itde-specific group.

```python
def ItdeOptionGroup(prefix, options):
    options = tuple(ItdeOption(prefix=prefix, **kwargs) for kwargs in options)
    return OptionGroup(prefix, options)
```